internal/controller/controllers/spoke_client_cache.go

```diff
@@ -40,40 +41,35 @@ func (c *spokeClientCache) Get(secret *corev1.Secret) (spoke_k8s_client.SpokeK8s
 	c.Lock()
 	defer c.Unlock()
 
-	// Get kubeconfig data and compute hash
-	kubeconfigData, err := c.getKubeconfigFromSecret(secret)
+	// Compute a hash for the content of the secret:
+	secretHash, err := c.calculateSecretHash(secret)
 	if err != nil {
 		return nil, err
 	}
-	kubeconfigHash := fmt.Sprintf("%x", sha256.New().Sum(kubeconfigData))
 
 	// Get client from cache or create a new one if not available
 	key := types.NamespacedName{Name: secret.Name, Namespace: secret.Namespace}
 	client, present := c.clientMap[key.String()]
-	if !present || client.kubeconfigHash != kubeconfigHash {
-		spokeK8sClient, err := c.clientFactory.CreateFromRawKubeconfig(kubeconfigData)
+	if !present || client.secretHash != secretHash {
+		spokeK8sClient, err := c.clientFactory.CreateFromSecret(secret)
```

Comment:
You need to remember to make CreateFromRawKubeconfig in the client factory private as you no longer need to export it.

Reply:
Actually I will remove it, is it just calls another method. Thanks Paul.

Reply:
Done.